### ISSUE TITLE
`-0` float literal interpreted as 0.0, should be -0.0

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#18246]
### Example
```nim
when true:
  import std/math
  proc fn(a: float) =
    echo (a, signbit(a))
  fn(0.0)
  fn(0)
  fn(-0.0)
  fn(-0) # bug
  var a: float = -0
  fn(a) # bug
```

### Current Output
(0.0, false)
(0.0, false)
(-0.0, true)
(0.0, false)
(0.0, false)

### Expected Output
(0.0, false)
(0.0, false)
(-0.0, true)
(-0.0, true)
(-0.0, true)

### Additional Information
1.5.1 6b97889f44d06f6649b6547a4ffdb18b94d6e578
happens for c, js, vm:
```nim
when true:
  import std/math
  proc fn(a: float) =
    echo (a, signbit(a))
  template main =
    fn(0.0)
    fn(0)
    fn(-0.0)
    fn(-0)
    var a: float = -0
    fn(a)
  static: main()
  main()
```
found this while writing https://github.com/nim-lang/Nim/pull/18248

**Comment by xflywind** [comment]
conversion from int to float loses information and vice versa.
```nim
when true:
  let x = -0
  echo float(x)
  let y = -0.0
  echo int(y)
```
output:
```
0.0
0
```

**Comment by Araq** [comment]
> conversion from int to float loses information and vice versa.

There is no negative 0 for integers...

**Comment by timotheecour** [comment]
> conversion from int to float loses information and vice versa.
> There is no negative 0 for integers...

yes but there shouldn't be a conversion from int to float in top example:
in `var a: float = -0` we have a literal `-0`, there is not conversion via float()

https://github.com/nim-lang/Nim/issues/18246#issuecomment-864955492 works as intended but IMO not the top example. Yes, it is a minor issue, easily fixable in user code by using -0.0 instead of -0, but still.

**Comment by metagn** [comment]
`-0'f` works, we could document that `-0` parses as `0`
